article/app/liveblog/KeyEventData.scala

```diff
@@ -0,0 +1,32 @@
+package liveblog
+
+import model.liveblog.BodyBlock.{KeyEvent, SummaryEvent}
+import model.liveblog.{Blocks, BodyBlock, LiveBlogDate}
+import org.scala_tools.time.Imports._
+
+object KeyEventData {
+
+  // just for convenience for use from the templates
+  def apply(maybeBlocks: Option[Blocks], timezone: DateTimeZone): Seq[KeyEventData] = {
+
+    val blocks = maybeBlocks.toSeq.flatMap(blocks => blocks.requestedBodyBlocks.getOrElse(Canonical.timeline, blocks.body))
+
+    apply(blocks, timezone)
+  }
+
+  def apply(blocks: Seq[BodyBlock], timezone: DateTimeZone): Seq[KeyEventData] = {
+
+    val TIMELINE_MAX_ENTRIES = 7
```

Comment:
`const` implies other things mutate no? Asking as I might be missing something completely.